dhctl/pkg/kubernetes/actions/converge/node.go

```diff
@@ -227,13 +228,27 @@ func WaitForNodesListBecomeReady(kubeCl *client.KubernetesClient, nodes []string
 			for _, c := range node.Status.Conditions {
 				if c.Type == apiv1.NodeReady {
 					if c.Status == apiv1.ConditionTrue {
-						readyNodes[node.Name] = struct{}{}
+						ready := true
+						if checker != nil {
+							var err error
+							ready, err = checker.IsReady(node.Name)
+							if err != nil {
+								log.WarnF("While doing check '%s' node %s has error: %v\n", checker.Name(), node.Name, err)
+							} else if !ready {
+								log.InfoF("Node %s is ready but %s is not ready\n", node.Name, checker.Name())
+							}
+						}
+
+						if ready {
+							readyNodes[node.Name] = struct{}{}
```

Comment:
why we don't use go_lib.set for this ?

Reply:
Because  `go_lib.set` belongs Deckhouse go module. Dhctl is different go module and Deckhouse depends on dhctl. If we use go_lib, then we get сircular dependency.